opsdroid/connector/matrix/connector.py

```diff
@@ -104,111 +103,136 @@ def _process_rooms_dict(self, rooms):
+            )
+            return
+
+        mapi.token = login_response.access_token
         mapi.sync_token = None
 
         for roomname, room in self.rooms.items():
-            response = await mapi.join_room(room["alias"])
-            self.room_ids[roomname] = response["room_id"]
+            response = await mapi.join(room["alias"])
+            if isinstance(response, nio.JoinError):
+                _LOGGER.error(
+                    f"Error while joining room: {room['alias']}, Message: {response.message} (status code {response.status_code})"
+                )
+
+            else:
+                self.room_ids[roomname] = response.room_id
+
         self.connection = mapi
 
         # Create a filter now, saves time on each later sync
-        self.filter_id = await self.make_filter(mapi)
+        self.filter_id = await self.make_filter(mapi, self.filter_json)
+        first_filter_id = await self.make_filter(
+            mapi, '{ "room": { "timeline" : { "limit" : 1 } } }'
+        )
 
         # Do initial sync so we don't get old messages later.
-        response = await self.connection.sync(
-            timeout_ms=3000,
-            filter='{ "room": { "timeline" : { "limit" : 1 } } }',
-            set_presence="online",
-        )
-        self.connection.sync_token = response["next_batch"]
+        response = await self.connection.sync(timeout=3000, sync_filter=first_filter_id)
+
+        if isinstance(response, nio.SyncError):
+            _LOGGER.error(
+                f"Error during initial sync: {response.message} (status code {response.status_code})"
+            )
+            return
+
+        self.connection.sync_token = response.next_batch
 
         if self.nick:
-            display_name = await self.connection.get_display_name(self.mxid)
+            display_name = await self.connection.get_displayname(self.mxid)
             if display_name != self.nick:
-                await self.connection.set_display_name(self.mxid, self.nick)
+                await self.connection.set_displayname(self.nick)
 
     async def disconnect(self):
         """Close the matrix session."""
-        await self.session.close()
+        await self.connection.close()
 
     async def _parse_sync_response(self, response):
-        self.connection.sync_token = response["next_batch"]
+        self.connection.sync_token = response.next_batch
 
         # Emit Invite events for every room in the invite list.
-        for roomid, room in response["rooms"]["invite"].items():
+        for roomid, roomInfo in response.rooms.invite.items():
             # Process the invite list to extract the person who invited us.
             invite_event = [
                 e
-                for e in room["invite_state"]["events"]
-                if "invite" == e.get("content", {}).get("membership")
+                for e in roomInfo.invite_state
+                if isinstance(e, nio.InviteMemberEvent)
+                if "invite" == e.membership
```

Comment:
This is real picky, but it' reads better this way round :grin: 
```suggestion
                if e.membership == "invite"
```